lib/bgs/service.rb

```diff
@@ -0,0 +1,197 @@
+    end
+
+    def with_multiple_attempts_enabled
+      attempt ||= 0
+      yield
+    rescue => e
+      attempt += 1
+      if attempt < MAX_ATTEMPTS
+        notify_of_service_exception(e, __method__.to_s, attempt, :warn)
+        retry
+      end
+
+      notify_of_service_exception(e, __method__.to_s)
+    end
+
+    def bgs_auth
+      {
+        jrn_dt: Time.current.iso8601,
+        jrn_lctn_id: Settings.bgs.client_station_id,
+        jrn_status_type_cd: 'U',
+        jrn_user_id: Settings.bgs.client_username,
+        jrn_obj_id: Settings.bgs.application,
+        ssn: @user[:ssn] # Just here to make the mocks work
+      }
+    end
+
+    def serialize_dependent_result(
+      participant,
+      participant_relationship_type,
+      family_relationship_type,
+      optional_fields = {}
+    )
+
+      {
+        vnp_participant_id: participant[:vnp_ptcpnt_id],
+        participant_relationship_type_name: participant_relationship_type,
+        family_relationship_type_name: family_relationship_type,
+        begin_date: optional_fields[:begin_date],
+        end_date: optional_fields[:end_date],
+        event_date: optional_fields[:event_date],
+        marriage_state: optional_fields[:marriage_state],
+        marriage_city: optional_fields[:marriage_city],
+        divorce_state: optional_fields[:divorce_state],
+        divorce_city: optional_fields[:divorce_city],
+        marriage_termination_type_code: optional_fields[:marriage_termination_type_code],
+        living_expenses_paid_amount: optional_fields[:living_expenses_paid],
+        type: optional_fields[:type]
+      }
+    end
+
+    def dependent_address(lives_with_vet, alt_address)
+      return @dependents_application.dig('veteran_contact_information', 'veteran_address') if lives_with_vet
+
+      alt_address
+    end
+
+    def generate_address(participant_id, address)
+      if address['view:lives_on_military_base'] == true
+        address['military_postal_code'] = address.delete('state_code')
+        address['military_post_office_type_code'] = address.delete('city')
+      end
+
+      create_address(@proc_id, participant_id, address)
+    end
+
+    def notify_of_service_exception(error, method, attempt = nil, status = :error)
+      msg = "Unable to #{method}: #{error.message}: try #{attempt} of #{MAX_ATTEMPTS}"
+      context = { icn: @user[:icn] }
+      tags = { team: 'vfs-ebenefits' }
+
+      return log_message_to_sentry(msg, :warn, context, tags) if status == :warn
+
+      log_exception_to_sentry(error, context, tags)
+      raise_backend_exception('BGS_686c_SERVICE_403', self.class, error)
+    end
+
+    def raise_backend_exception(key, source, error)
+      exception = BGS::ServiceException.new(
+        key,
+        { source: source.to_s },
+        403,
+        error.message
+      )
+
+      raise exception
+    end
+
+    def format_date(date)
+      return nil if date.nil?
+
+      Date.parse(date).to_time.iso8601
+    end
```

Comment:
Probably nitpicky, but this is duplicate code 

https://github.com/department-of-veterans-affairs/vets-api/pull/4546/files#diff-39a641783cd9a07c7cf96358929c6680.

Maybe create a `lib/bgs/date_formatter.rb` mixin?